exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1805 @@
+	params.Set("timestamp", strconv.Itoa(int(timestamp)))
+	params.Set("recvWindow", strconv.Itoa(30000))
+	var resp []OrderRateLimit
+	return resp, b.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary, http.MethodGet, orderRateLimit, params, spotDefaultRate, &resp)
+}
+
+// NewOrder sends a new order to Binance
+func (b *Binanceus) NewOrder(ctx context.Context, o *NewOrderRequest) (NewOrderResponse, error) {
+	var resp NewOrderResponse
+	if err := b.newOrder(ctx, orderRequest, o, &resp); err != nil {
+		return resp, err
+	}
+	if resp.Code != 0 {
+		return resp, errors.New(resp.Msg)
+	}
+	return resp, nil
+}
+
+// NewOrderTest sends a new test order to Binanceus
+// to test new order creation and signature/recvWindow long. The endpoint creates and validates a new order but does not send it into the matching engine.
+func (b *Binanceus) NewOrderTest(ctx context.Context, o *NewOrderRequest) (*NewOrderResponse, error) {
+	var resp NewOrderResponse
+	return &resp, b.newOrder(ctx, testCreateNeworder, o, &resp)
+}
+
+// newOrder
+func (b *Binanceus) newOrder(ctx context.Context, api string, o *NewOrderRequest, resp *NewOrderResponse) error {
+	params := url.Values{}
+	symbol, err := b.FormatSymbol(o.Symbol, asset.Spot)
+	if err != nil {
+		return err
+	}
+	params.Set("symbol", symbol)
+	params.Set("side", o.Side)
+	params.Set("type", string(o.TradeType))
+	if o.QuoteOrderQty > 0 {
+		params.Set("quoteOrderQty", strconv.FormatFloat(o.QuoteOrderQty, 'f', -1, 64))
+	} else {
+		params.Set("quantity", strconv.FormatFloat(o.Quantity, 'f', -1, 64))
+	}
+	if o.TradeType == BinanceRequestParamsOrderLimit {
+		params.Set("price", strconv.FormatFloat(o.Price, 'f', -1, 64))
+	}
+	if o.TimeInForce != "" {
+		params.Set("timeInForce", string(o.TimeInForce))
+	}
+	if o.NewClientOrderID != "" {
+		params.Set("newClientOrderId", o.NewClientOrderID)
+	}
+	if o.StopPrice != 0 {
+		params.Set("stopPrice", strconv.FormatFloat(o.StopPrice, 'f', -1, 64))
+	}
+	if o.IcebergQty != 0 {
+		params.Set("icebergQty", strconv.FormatFloat(o.IcebergQty, 'f', -1, 64))
+	}
+	if o.NewOrderRespType != "" {
+		params.Set("newOrderRespType", o.NewOrderRespType)
+	}
+	return b.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodPost, api, params,
+		spotOrderRate, resp)
+}
+
+// GetOrder to check a trade order's status.
+func (b *Binanceus) GetOrder(ctx context.Context, arg OrderRequestParams) (*Order, error) {
+	var resp Order
+	params := url.Values{}
+	if arg.Symbol == "" {
+		return nil, errIncompleteArguments
+	}
+	params.Set("symbol", strings.ToUpper(arg.Symbol))
+	if arg.OrderID > 0 {
+		params.Set("orderId", fmt.Sprintf("%d", arg.OrderID))
+	}
+	timestamp := time.Now().UnixMilli()
+	params.Set("timestamp", fmt.Sprintf("%d", timestamp))
+	if arg.OrigClientOrderId != "" {
+		params.Set("origClientOrderId", arg.OrigClientOrderId)
+	}
+	if arg.RecvWindow > 200 && arg.RecvWindow <= 6000 {
+		params.Set("recvWindow", fmt.Sprint(arg.RecvWindow))
```

Comment:
I believe using strconv package is more optimized compared to fmt package. Let me know your thoughts on changed all these over.